lib/Sema/CSSimplify.cpp

```diff
@@ -3056,11 +3056,19 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
   if (func1->getGlobalActor() || func2->getGlobalActor()) {
     if (func1->getGlobalActor() && func2->getGlobalActor()) {
       // If both have a global actor, match them.
-      TypeMatchOptions subflags = getDefaultDecompositionOptions(flags);
-      auto result = matchTypes(func1->getGlobalActor(), func2->getGlobalActor(),
-                               ConstraintKind::Equal, subflags, locator);
-      if (result == SolutionKind::Error)
-        return getTypeMatchFailure(locator);
+      const auto subflags = getDefaultDecompositionOptions(flags);
+      auto result = [&]() {
+        llvm::SaveAndRestore<ConstraintSystemOptions> options(
+            Options, Options - ConstraintSystemFlags::AllowFixes);
```

Comment:
This kind of seems like it's trying to workaround some problem and I think that problem is that `locator` passed to the `matchType` under this declaration should actually be different (with an extract element at the end) to identify for `repairFailures` that is going to be called by upcoming `matchTypes` that we are matching an attribute location.

Reply:
Yeah, I have actually considered this option before. Just fixed!